### PR TITLE
Server Message Handler Safety

### PR DESCRIPTION
**Pull request by geoffthemedio**
Set a default message size limit and put a try-catch around message buffer allocation. Should help with #2967

### CODE REVIEW

File: server/dmain.cpp [hunk 1]
```diff
@@ -68,7 +68,7 @@ int wmain(int argc, wchar_t* argv[], wchar_t* envp[]) {
         GetOptionsDB().Add<int>("network.server.turn-timeout.max-interval",             UserStringNop("OPTIONS_DB_TIMEOUT_INTERVAL"),           0);
         GetOptionsDB().Add<bool>("network.server.turn-timeout.fixed-interval",          UserStringNop("OPTIONS_DB_TIMEOUT_FIXED_INTERVAL"),     false);
         GetOptionsDB().Add<std::string>("setup.game.uid",                               UserStringNop("OPTIONS_DB_GAMESETUP_UID"),              "");
-        GetOptionsDB().Add<int>("network.server.client-message-size.max",               UserStringNop("OPTIONS_DB_CLIENT_MESSAGE_SIZE_MAX"),    0);
+        GetOptionsDB().Add<int>("network.server.client-message-size.max",               UserStringNop("OPTIONS_DB_CLIENT_MESSAGE_SIZE_MAX"),    1000000);
```
Comment (geoffthemedio) [inline server/dmain.cpp:71]: Not sure about this default. I have a "big" game where I let the AIs play for 200+ turn, and they send orders messages of about 140 000 bytes, which is well under 1 000 000, but is it plausible players would send even bigger messages?

**Comment by geoffthemedio** [comment]
pushed the try-catch part into master

**Comment by o01eg** [comment]
Won't it be pushed to 0.4.10?

**Comment by Vezzra** [comment]
Um, closed as what? Merged, invalid, abandoned? If the first, should the changes that did get pushed master be cherry picked to the release branch?

If yes, please assign to the 0.4.10 milestone and tag with "cherry-pick for release". If no, please assign to the post release milestone.

**Comment by geoffthemedio** [comment]
> Um, closed as what?

not sure, see:

> pushed the try-catch part into master

**Comment by Vezzra** [comment]
Well, that's the major part, so I guess closed as merged is the most fitting option here.

**Pull request closed without merging**